client/src/main/java/com/arcbees/beeshop/client/application/gin/BootstrapperImpl.java

```diff
@@ -22,47 +22,50 @@
 import com.gwtplatform.mvp.client.Bootstrapper;
 import com.gwtplatform.mvp.client.proxy.PlaceManager;
 import com.gwtplatform.mvp.shared.proxy.PlaceRequest;
+import com.gwtplatform.mvp.shared.proxy.TokenFormatter;
 
 public class BootstrapperImpl implements Bootstrapper {
     private final PlaceManager placeManager;
     private final LocaleHelper localeHelper;
+    private final TokenFormatter tokenFormatter;
 
     @Inject
     BootstrapperImpl(
             PlaceManager placeManager,
-            LocaleHelper localeHelper) {
+            LocaleHelper localeHelper,
+            TokenFormatter tokenFormatter) {
         this.placeManager = placeManager;
         this.localeHelper = localeHelper;
+        this.tokenFormatter = tokenFormatter;
     }
 
     @Override
     public void onBootstrap() {
-        PlaceRequest currentPlaceRequest = placeManager.getCurrentPlaceRequest();
-        String nameToken = Strings.nullToEmpty(History.getToken());
+        String historyToken = History.getToken();
+        PlaceRequest currentPlaceRequest =
+                tokenFormatter.toPlaceRequest(historyToken.isEmpty() ? NameTokens.HOME : historyToken);
+        String nameToken = currentPlaceRequest.getNameToken();
 
         if (localeHelper.isEnglish()) {
             if (!Strings.isNullOrEmpty(nameToken) && !NameTokens.isEnglish(nameToken)) {
```

Comment:
`nameToken` shouldn't be null or empty now, and the content of the conditions is the same. You could create a method like this
```
private boolean localeAndNameTokensAreNotSame(nameToken) {
    return (localeHelper.isEnglish() && !NameTokens.isEnglish(nameToken))
        || (localeHelper.isFrench() && NameTokens.isEnglish(nameToken));
}
```